drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -15,13 +16,56 @@
 #include "sl_net_constants.h"
 #include "sl_wifi_types.h"
 #include "sl_wifi_callback_framework.h"
+#include "sl_net_default_values.h"
 #include "sl_wifi.h"
 #include "sl_net.h"
 
 LOG_MODULE_REGISTER(siwx91x_wifi);
 
 NET_BUF_POOL_FIXED_DEFINE(siwx91x_tx_pool, 1, _NET_ETH_MAX_FRAME_SIZE, 0, NULL);
 
+static inline int siwx91x_sl_to_z_mode(sl_wifi_interface_t interface)
+{
+	switch (interface) {
+	case SL_WIFI_CLIENT_INTERFACE:
+		return WIFI_STA_MODE;
+	case SL_WIFI_AP_INTERFACE:
+		return WIFI_AP_MODE;
+	default:
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static inline int siwx91x_bandwidth(enum wifi_frequency_bandwidths bandwidth)
+{
+	switch (bandwidth) {
+	case WIFI_FREQ_BANDWIDTH_20MHZ:
+		return SL_WIFI_BANDWIDTH_20MHz;
+	case WIFI_FREQ_BANDWIDTH_40MHZ:
+		return SL_WIFI_BANDWIDTH_40MHz;
+	case WIFI_FREQ_BANDWIDTH_80MHZ:
+		return SL_WIFI_BANDWIDTH_80MHz;
+	default:
+		return -EINVAL;
+	}
+}
+
+static inline int siwx91x_security(enum wifi_security_type security)
```

Comment:
Using `inline` for all of these smells like premature optimization. I think it'd better to add `inline` very conservatively, only when you've actually done some profiling and concluded that the compiler isn't doing the correct thing otherwise. Another problem with `inline` is that you suppress all warnings/errors of potentially unused functions (`static inline` is a common thing in header files, so it's normal for such functions to be declared but not used, however this is not the case for c-files like here)